src/SprykerEco/Yves/Payone/Checkout/Process/StepFactory.php

```diff
@@ -0,0 +1,241 @@
+use SprykerEco\Yves\Payone\PayoneDependencyProvider;
+use SprykerEco\Yves\Payone\Plugin\Provider\PayoneControllerProvider;
+
+class StepFactory extends AbstractFactory
+{
+
+    /**
+     * @param \Spryker\Yves\StepEngine\Process\StepCollectionInterface $stepCollection
+     *
+     * @return \Spryker\Yves\StepEngine\Process\StepEngine
+     */
+    public function createStepEngine(StepCollectionInterface $stepCollection)
+    {
+        return new StepEngine($stepCollection, $this->createDataContainer());
+    }
+
+    /**
+     * @return \Spryker\Yves\StepEngine\Process\StepCollection
+     */
+    public function createStepCollection()
+    {
+        $stepCollection = new StepCollection(
+            $this->getUrlGenerator(),
+            PayoneControllerProvider::EXPRESS_CHECKOUT_FAILURE
+        );
+
+        $stepCollection
+            ->addStep($this->createInitExpressCheckoutStep())
+            ->addStep($this->createLoadExpressCheckoutDetailsStep())
+            ->addStep($this->createSummaryStep())
+            ->addStep($this->createPlaceOrderStep())
+            ->addStep($this->createSuccessStep());
+
+        return $stepCollection;
+    }
+
+    /**
+     * @return \SprykerEco\Client\Payone\PayoneClientInterface
+     */
+    public function getPayoneClient()
+    {
+        return $this->getProvidedDependency(PayoneDependencyProvider::CLIENT_PAYONE);
+    }
+
+    /**
+     * @return \Spryker\Client\Shipment\ShipmentClientInterface
+     */
+    public function getShipmentClient()
+    {
+        return $this->getProvidedDependency(PayoneDependencyProvider::CLIENT_SHIPMENT);
+    }
+
+    /**
+     * @return \SprykerEco\Yves\Payone\Handler\ExpressCheckout\QuoteHydrator
+     */
+    public function createQuoteHydrator()
+    {
+        return new QuoteHydrator(
+            $this->getShipmentClient(),
+            $this->getCustomerClient()
+        );
+    }
+
+    /**
+     * @return \Spryker\Yves\Checkout\DataContainer\DataContainer
+     */
+    protected function createDataContainer()
+    {
+        return new DataContainer(new CheckoutToQuoteBridge($this->getQuoteClient()));
+    }
+
+    /**
+     * @return \Spryker\Client\Quote\QuoteClientInterface
+     */
+    protected function getQuoteClient()
+    {
+        return $this->getProvidedDependency(PayoneDependencyProvider::CLIENT_QUOTE);
+    }
+
+    /**
+     * @return \Symfony\Component\Routing\Generator\UrlGeneratorInterface
+     */
+    protected function getUrlGenerator()
+    {
+        return $this->getApplication()['url_generator'];
+    }
+
+    /**
+     * @return mixed
+     */
+    protected function getApplication()
+    {
+        return $this->getProvidedDependency(PayoneDependencyProvider::PLUGIN_APPLICATION);
+    }
+
+    /**
+     * @return \Spryker\Yves\ProductBundle\Grouper\ProductBundleGrouper
+     */
+    protected function createProductBundleGrouper()
+    {
+        return new ProductBundleGrouper();
```

Comment:
Please avoid this.
It couples code with product bundles.
For AmazonPay we've moved code to project level